modules/core/04-channel/keeper/upgrade.go

```diff
@@ -85,6 +88,36 @@ func (k Keeper) WriteUpgradeTryChannel(
 	// grab channel inside this function to get most current channel status
 }
 
+// validateUpgradeFields validates the proposed upgrade fields against the existing channel.
+// It returns an error if the following constraints are not met:
+// - there exists at least one valid proposed change to the existing channel fields
+// - the proposed order is a subset of the existing order
+// - the proposed connection hops do not exist
+// - the proposed version is non-empty (checked in UpgradeFields.ValidateBasic())
+// - the proposed connection hops are not open
+func (k Keeper) validateUpgradeFields(ctx sdk.Context, proposedUpgrade types.UpgradeFields, currentChannel types.Channel) error {
```

Comment:
shoud we add the [`proposedConnectionHops` checks](https://github.com/cosmos/ibc-go/pull/3628#discussion_r1210307663) to this validation function?

Reply:
There are three types of validation we can do on an upgrade:
- stateless validation (upgrade.ValidateBasic())
- stateful validation of self upgrade
- stateful validation of self upgrade relative to counterparty upgrade

The proposed connection hop check falls into the third category, which is not the aim of this function. This function's purpose is to do stateful validation of the self upgrade